Core/DistortionPointHandler.cs

```diff
@@ -0,0 +1,135 @@
+
+    public delegate float DistortionProgressFunction(float currentProgress, int ticksPassed);
+
+    public delegate bool DistortionActiveFunction(float currentProgress, float currentIntensity, int ticksPassed);
+
+    public class DistortionPoint
+    {
+        public Vector2 Position;
+
+        public bool Active = true;
+
+        public float Intensity;
+        public float Progress;
+
+        public int TicksPassed = 0;
+
+        public DistortionIntensityFunction UpdateIntensity;
+        public DistortionProgressFunction UpdateProgress;
+        public DistortionActiveFunction UpdateActive;
+
+        public DistortionPoint(Vector2 position, float intensity, float progress, DistortionIntensityFunction updateIntensity, DistortionProgressFunction updateProgress, DistortionActiveFunction updateActive)
+        {
+            Position = position;
+            Intensity = intensity;
+            Progress = progress;
+            UpdateIntensity = updateIntensity;
+            UpdateProgress = updateProgress;
+            UpdateActive = updateActive;
+        }
+    }
+    public class DistortionPointHandler : ModSystem
+    {
+        public static List<DistortionPoint> DistortionPoints = new List<DistortionPoint>();
+
+        public override void Load()
+        {
+            if (Main.netMode != NetmodeID.Server)
+            {
+                Ref<Effect> screenRef = new Ref<Effect>(Mod.Assets.Request<Effect>("Effects/DistortionPulse").Value);
+                Filters.Scene["DistortionPulse"] = new Filter(new ScreenShaderData(screenRef, "MainPS"), EffectPriority.VeryHigh);
+                Filters.Scene["DistortionPulse"].Load();
+            }
+        }
+
+        public override void PostUpdateProjectiles()
+        {
+            if (Main.gameMenu)
+                return;
+
+            int projectilesFound = 0;
+            int numberOfShockwaves = 0;
+            float[] progresses = new float[10];
+            float[] intensity = new float[10];
+            Vector2[] positions = new Vector2[10];
+
+            foreach (DistortionPoint point in DistortionPoints.ToArray())
+            {
+                if (!point.Active)
+                    DistortionPoints.Remove(point);
+            }
+            foreach (DistortionPoint point in DistortionPoints)
+            {
+                point.Intensity = point.UpdateIntensity.Invoke(point.Intensity, ++point.TicksPassed);
+                point.Progress = point.UpdateProgress.Invoke(point.Progress, point.TicksPassed);
+                point.Active = point.UpdateActive.Invoke(point.Progress, point.Intensity, point.TicksPassed);
+
+                intensity[projectilesFound] = point.Intensity;
+                positions[projectilesFound] = point.Position;
+                progresses[projectilesFound] = point.Progress;
+                numberOfShockwaves++;
+                projectilesFound++;
```

Comment:
this is a misleading name -- distortion points are not projectiles. What if a non-projectile adds a distortion point?

Reply:
Fixed